src/qz/ws/PrintSocketClient.java

```diff
@@ -324,7 +324,8 @@ private void processMessage(Session session, JSONObject json, SocketConnection c
                 SerialProperties props = new SerialProperties(params.optJSONObject("properties"));
                 SerialIO serial = connection.getSerialPort(params.optString("port"));
                 if (serial != null) {
-                    serial.sendData(props, params.optString("data"));
+                    JSONObject data = params.optJSONObject("data");
+                    serial.sendData(props, SerialUtilities.getDataTypeJSON(data), data.optString("data"));
```

Comment:
Accounting for string types here - if optJSONObject returns null, we need to pull the command from params[data] as a string (like before) instead of from data[data]